platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/datasource/VirtualDataSource.java

```diff
@@ -83,9 +84,9 @@ default void saveRecords(
         saveRecords(
                 firstLeafPath,
                 lastLeafPath,
-                pathHashRecordsToUpdate,
-                leafRecordsToAddOrUpdate,
-                leafRecordsToDelete,
+                pathHashRecordsToUpdate.toList(),
+                leafRecordsToAddOrUpdate.toList(),
+                leafRecordsToDelete.toList(),
```

Comment:
We copy each stream to an array. Can we get rid of streams and produce lists at the source?